src/codegen/compile.ml

```diff
@@ -329,9 +341,8 @@ module SR = struct
     | _ -> t1 = t2
 
   let to_var_type : t -> value_type = function
-    | Vanilla -> I32Type
+    | Vanilla -> I64Type
     | UnboxedWord64 _ -> I64Type
-    | UnboxedWord32 _ -> I32Type
```

Comment:
I would have been tempted to keep this so all arithmetic on unboxed upto 32-bit values just stayed the same, with the only changes happening on entry-to / exit-from stack but I guess you removed this long before I made the tagging changes....

Reply:
Yes, I removed it long time ago. I believe you are right. Having it kept would have simplified the redesign of the arithmetic operations that use 32-bit to 64-bit extension to detect some overflows etc. Probably, it does not make a big difference w.r.t. performance if the Wasm stack is anyways 64-bit aligned internally and IC charging is not higher for 64-bit instruction arithmetics than 32-bit. I can keep it on my backlog of potential improvement or analysis of optimization, if this is okay.